apps/modernization-ui/src/design-system/richTooltip/RichTooltip.spec.tsx

```diff
@@ -15,4 +16,72 @@ describe('when a rich tooltip is displayed', () => {
+        const MockRichTooltipAnchorElement = () => {
+            const mockRichTooltipAnchorRef = useRef<HTMLDivElement>(null);
+            return (
+                <div>
+                    <div ref={mockRichTooltipAnchorRef} data-testid="mock-rich-tooltip-anchor-div">
+                        <p>Mock Anchor Element for Tooltip</p>
+                    </div>
+                    <div>
+                        <RichTooltip anchorRef={mockRichTooltipAnchorRef}>
+                            <p>Tooltip Content Text</p>
+                        </RichTooltip>
+                    </div>
+                </div>
+            );
+        };
+
+        const { container, getByTestId } = render(<MockRichTooltipAnchorElement />);
+        fireEvent.mouseEnter(getByTestId('mock-rich-tooltip-anchor-div'));
+        expect(container).toBeTruthy();
+    });
+
+    it('should properly display the children props passed into the component.', async () => {
+        const MockRichTooltipAnchorElement = () => {
+            const mockRichTooltipAnchorRef = useRef<HTMLDivElement>(null);
+            return (
+                <div>
+                    <div ref={mockRichTooltipAnchorRef} data-testid="mock-rich-tooltip-anchor-div">
+                        <p>Mock Anchor Element for Tooltip</p>
+                    </div>
+                    <div>
+                        <RichTooltip anchorRef={mockRichTooltipAnchorRef}>
+                            <p>Tooltip Content Text</p>
+                        </RichTooltip>
+                    </div>
+                </div>
+            );
+        };
+
+        const { getByTestId, findByText } = render(<MockRichTooltipAnchorElement />);
+        fireEvent.mouseEnter(getByTestId('mock-rich-tooltip-anchor-div'));
+        expect(await findByText('Tooltip Content Text')).toBeInTheDocument();
+        expect(await findByText('Tooltip Content Text')).toBeVisible();
+    });
+
+    it('should not display the children props passed into the component when the mouse is not over the tooltip.', async () => {
+        const MockRichTooltipAnchorElement = () => {
+            const mockRichTooltipAnchorRef = useRef<HTMLDivElement>(null);
+            return (
+                <div>
+                    <div ref={mockRichTooltipAnchorRef} data-testid="mock-rich-tooltip-anchor-div">
+                        <p>Mock Anchor Element for Tooltip</p>
+                    </div>
+                    <div data-testid="mock-rich-tooltip-container-div">
+                        <RichTooltip anchorRef={mockRichTooltipAnchorRef}>
+                            <p>Tooltip Content Text</p>
+                        </RichTooltip>
+                    </div>
+                </div>
+            );
+        };
+
+        const { getByTestId } = render(<MockRichTooltipAnchorElement />);
+        fireEvent.mouseEnter(getByTestId('mock-rich-tooltip-anchor-div'));
```

Comment:
I would suggest using userEvent as this is what we have been using across unit tests as fireEvent dispatches DOM events, whereas user-event simulates full interactions, which may fire multiple events and do additional checks along the way.

Reply:
Good idea! thanks for the info